core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -565,10 +583,11 @@ public void forceAvailabilityMode(String cacheName, AvailabilityMode availabilit
       public void handleViewChange(final ViewChangedEvent e) {
          // Need to recover existing caches asynchronously (in case we just became the coordinator).
          // Cannot use the async notification thread pool, by default it only has 1 thread.
-         asyncTransportExecutor.submit(new Runnable() {
+         viewHandlingCompletionService.submit(new Callable<Void>() {
```

Comment:
I was thinking that having an extra null parameter is no better than `return null`. Or is there any other reason not to use `Callable<Void>`?

And yeah, I was thinking about the other `submit()` call for merge, with uses a local variable that's a `CompletionService`.

Anyway, I'm replacing the completion service with an executor service, so I'll have a `void submit(Runnable)` option. Running the tests now...